Parser does not handle backslash in raw strings

The usage of the following snippet in the code would lead to failures when trying to create SemanticDB. 
```
 val host = "foo"
 val share = "bar"
 raw"""\\$host\$share\"""
```
It looks like the last backlash is treated by the parser as an escape for char `"`, which then leads to treating the rest of the source files as an unclosed string. In the result, if we would try to define some other interpreted strings parser would fail to process them resulting in the compiler failure, here's a stack trace for my use case: 
```
2021.09.06 10:28:48 ERROR failed to generate semanticdb for ...scala/scala-native/javalib/src/main/scala/scala/scalanative/nio/fs/windows/WindowsPathParser.scala:
.../scala/scalanative/nio/fs/windows/WindowsPathParser.scala:138: error: unclosed multi-line string interpolation
          s"Illegal character `${segment.charAt(invalidCharIdx)}` in path",
                                                                ^
	at scala.meta.internal.tokenizers.Reporter.incompleteInputError(Reporter.scala:27)
	at scala.meta.internal.tokenizers.Reporter.incompleteInputError$(Reporter.scala:26)
	at scala.meta.internal.tokenizers.Reporter$$anon$1.incompleteInputError(Reporter.scala:33)
	at scala.meta.internal.tokenizers.Reporter.incompleteInputError(Reporter.scala:29)
	at scala.meta.internal.tokenizers.Reporter.incompleteInputError$(Reporter.scala:29)
	at scala.meta.internal.tokenizers.Reporter$$anon$1.incompleteInputError(Reporter.scala:33)
	at scala.meta.internal.tokenizers.LegacyScanner.getStringPart(LegacyScanner.scala:762)
	at scala.meta.internal.tokenizers.LegacyScanner.fetchStringPart(LegacyScanner.scala:788)
	at scala.meta.internal.tokenizers.LegacyScanner.fetchToken(LegacyScanner.scala:321)
	at scala.meta.internal.tokenizers.LegacyScanner.nextToken(LegacyScanner.scala:251)
	at scala.meta.internal.tokenizers.LegacyScanner.foreach(LegacyScanner.scala:1111)
	at scala.meta.internal.tokenizers.ScalametaTokenizer.uncachedTokenize(ScalametaTokenizer.scala:157)
	at scala.meta.internal.tokenizers.ScalametaTokenizer.$anonfun$tokenize$1(ScalametaTokenizer.scala:17)
	at scala.collection.concurrent.TrieMap.getOrElseUpdate(TrieMap.scala:895)
	at scala.meta.internal.tokenizers.ScalametaTokenizer.tokenize(ScalametaTokenizer.scala:17)
	at scala.meta.internal.tokenizers.ScalametaTokenizer$$anon$2.apply(ScalametaTokenizer.scala:336)
	at scala.meta.tokenizers.Api$XtensionTokenizeDialectInput.tokenize(Api.scala:25)
	at scala.meta.tokenizers.Api$XtensionTokenizeInputLike.tokenize(Api.scala:14)
	at scala.meta.internal.parsers.ScalametaParser.scannerTokens$lzycompute(ScalametaParser.scala:241)
	at scala.meta.internal.parsers.ScalametaParser.scannerTokens(ScalametaParser.scala:241)
	at scala.meta.internal.parsers.ScalametaParser.<init>(ScalametaParser.scala:155)
	at scala.meta.internal.parsers.ScalametaParser$$anon$263.apply(ScalametaParser.scala:5257)
	at scala.meta.parsers.Api$XtensionParseDialectInput.parse(Api.scala:25)
	at scala.meta.internal.semanticdb.scalac.ParseOps$XtensionCompilationUnitSource.toSource(ParseOps.scala:17)
	at scala.meta.internal.semanticdb.scalac.TextDocumentOps$XtensionCompilationUnitDocument.toTextDocument(TextDocumentOps.scala:201)
	at scala.meta.internal.semanticdb.scalac.TextDocumentOps$XtensionCompilationUnitDocument.toTextDocument(TextDocumentOps.scala:52)
	at scala.meta.internal.semanticdb.scalac.SemanticdbPipeline$SemanticdbTyperComponent$ComputeSemanticdbPhase.saveSemanticdbForCompilationUnit(SemanticdbPipeline.scala:60)
	at scala.meta.internal.semanticdb.scalac.SemanticdbPipeline$SemanticdbTyperComponent$ComputeSemanticdbPhase.apply(SemanticdbPipeline.scala:67)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:453)

```

--- a/scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala
+++ b/scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala
@@ -695,10 +695,13 @@ class LegacyScanner(input: Input, dialect: Dialect) {
         setStrVal()
         token = STRINGLIT
       }
-    } else if (ch == '\\') {
-      nextRawChar()
+    } else if (ch == '\\' && !multiLine) {
       putChar(ch)
       nextRawChar()
+      if (ch == '"' || ch == '\\') {
+        putChar(ch)
+        nextRawChar()
+      }
       getStringPart(multiLine)
     } else if (ch == '$' && !isUnicodeEscape) {
       if (!getDollar()) {

--- a/tests/shared/src/test/scala/scala/meta/tests/parsers/TermSuite.scala
+++ b/tests/shared/src/test/scala/scala/meta/tests/parsers/TermSuite.scala
@@ -806,7 +806,7 @@ class TermSuite extends ParseSuite {
   test("fstring-interpolation") {
     val Term.Interpolate(
       Term.Name("f"),
-      List(Lit.String("\\u"), Lit.String("%04x")),
+      List(Lit.String("\\\\u"), Lit.String("%04x")),
       List(Term.Name("oct"))
     ) = term("""f"\\u$oct%04x"""")
   }

--- a/tests/shared/src/test/scala/scala/meta/tests/tokenizers/TokenizerSuite.scala
+++ b/tests/shared/src/test/scala/scala/meta/tests/tokenizers/TokenizerSuite.scala
@@ -1008,7 +1008,7 @@ class TokenizerSuite extends BaseTokenizerSuite {
       BOF(),
       Interpolation.Id("s"),
       Interpolation.Start(),
-      Interpolation.Part("\"Hello\", "),
+      Interpolation.Part("\\\"Hello\\\", "),
       Interpolation.SpliceStart(),
       Ident("person"),
       Interpolation.SpliceEnd(),
@@ -1021,6 +1021,27 @@ class TokenizerSuite extends BaseTokenizerSuite {
     assert(
       ("""s"\\"Hello"""").tokenize.isInstanceOf[Tokenized.Error]
     )
+
+  }
+
+  test("Multiline interpolated string - ignore escape") {
+    val Tokens(
+      BOF(),
+      Interpolation.Id("raw"),
+      Interpolation.Start(),
+      Interpolation.Part("\\"),
+      Interpolation.SpliceStart(),
+      Ident("host"),
+      Interpolation.SpliceEnd(),
+      Interpolation.Part("\\"),
+      Interpolation.SpliceStart(),
+      Ident("share"),
+      Interpolation.SpliceEnd(),
+      Interpolation.Part("\\"),
+      Interpolation.End(),
+      EOF()
+    ) =
+      ("raw\"\"\"\\$host\\$share\\\"\"\"").tokenize.get
 
   }
 }

